extensions/amazon-sam/runtime/src/main/java/io/quarkus/amazon/sam/SamHandler.java

```diff
@@ -0,0 +1,192 @@
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.core.HttpHeaders;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.serverless.proxy.AwsProxyExceptionHandler;
+import com.amazonaws.serverless.proxy.internal.SecurityUtils;
+import com.amazonaws.serverless.proxy.model.AwsProxyRequest;
+import com.amazonaws.serverless.proxy.model.AwsProxyResponse;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.module.afterburner.AfterburnerModule;
+
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.DefaultLastHttpContent;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.ReferenceCountUtil;
+import io.quarkus.netty.runtime.virtual.VirtualClientConnection;
+import io.quarkus.runtime.Application;
+import io.quarkus.vertx.http.runtime.VertxHttpRecorder;
+
+@SuppressWarnings("unused")
+public class SamHandler implements RequestStreamHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(SamHandler.class);
+
+    private static ObjectMapper objectMapper = new ObjectMapper();
+    private ObjectReader objectReader = objectMapper.readerFor(AwsProxyRequest.class);
+    private ObjectWriter objectWriter = objectMapper.writerFor(AwsProxyResponse.class);
+
+    static {
+        objectMapper.registerModule(new AfterburnerModule());
+    }
+
+    static {
+        StringWriter error = new StringWriter();
+        PrintWriter errorWriter = new PrintWriter(error, true);
+        if (Application.currentApplication() == null) {
+            try {
+                Class appClass = Class.forName("io.quarkus.runner.ApplicationImpl1");
+                String[] args = {};
+                Application app = (Application) appClass.newInstance();
+                app.start(args);
+                errorWriter.println("Quarkus bootstrapped successfully.");
+            } catch (Exception ex) {
+                errorWriter.println("Quarkus bootstrap failed.");
+                ex.printStackTrace(errorWriter);
+            }
+        } else {
+            errorWriter.println("Quarkus bootstrapped successfully.");
+        }
+    }
+
+    private AwsProxyExceptionHandler exceptionHandler = new AwsProxyExceptionHandler();
+
+    @Override
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
```

Comment:
Can you give a bit of explanation as to why the AWS proxy implementation is mixed in with the SAM implementation? As far as I can tell from the AWS docs they are really seperate concepts so I don't really understand why this is all mixed together?